modules/grpc/pubsub/pubsub-dest-worker.cpp

```diff
@@ -99,30 +99,70 @@ DestWorker::insert(LogMessage *msg)
 {
   DestDriver *owner_ = this->get_owner();
 
-  ScratchBuffersMarker m;
-  GString *buf = scratch_buffers_alloc_and_mark(&m);
-  Slice buf_slice;
   size_t message_bytes = 0;
 
   ::google::pubsub::v1::PubsubMessage *message = this->request.add_messages();
 
-  buf_slice = this->format_template(owner_->data, msg, buf, NULL, this->super->super.seq_num);
-  message->set_data(buf_slice.str, buf_slice.len);
-  message_bytes += buf_slice.len;
-
-  auto attributes = message->mutable_attributes();
-  for (const auto &attribute : owner_->attributes)
+  if (owner_->protovar)
```

Comment:
[optional]

I think this is a good candidate for factoring out to some smaller functions.